code/game/objects/items/weapons/grenades/custom_grenades.dm

```diff
@@ -303,3 +303,79 @@
 	new /obj/item/multitool(src) // used to adjust the chemgrenade's signaller
 	new /obj/item/assembly/signaler(src)
 	return
+
+// --------------------------------------
+//  BSA Signaller
+// --------------------------------------
+
+/obj/item/grenade/bsa
+	name = "bluespace artillery signal beacon"
+	desc = "Call in a one time strike with bluespace artillery. It is filled with various biometric security measures."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "batterer"
+	item_state = "batterer"
+	req_one_access = list(access_cent_commander)
+	var/strike_init_time = 8 //in seconds, we multiply by 10 below so we can use this var in the messages we send. so a value of 8 here would be a value of 80 for the timer.
+	var/caller
+	var/centcom_cancast = FALSE //incase admin wants to varedit to allow on Z 2 for whatever reason.
+
+/obj/item/grenade/bsa/attack_self()
+	var/mob/user = usr
+	var/turf/bombturf = get_turf(src)
+	if(active)
+		return
+
+	if(is_admin_level(bombturf.z) && !centcom_cancast)
+		to_chat(user, "<span class = 'warning'>The strike craft is not authorized to fire on that Z level!</span>")
+		return
+
+	if((!ismindshielded(user)) || ((user.mind.special_role != SPECIAL_ROLE_ERT) && (user.mind.special_role != SPECIAL_ROLE_DEATHSQUAD) && (!allowed(user)))) //if is not mindshielded, or is not an ERT/DS AND does not have high level CC access. naval officers have access_cent_commander, but not special role so we don't want to lock them out.
+		to_chat(user, "<span class = 'warning'>Unauthorized user detected.</span>")
+	else
+		for(var/obj/item/grenade/bsa/B in world) //check if anyone else is calling a strike currently, we don't want to spam the announcements
+			if(B.active && B != src)
+				to_chat(user, "<span class = 'warning'>Another strike is already in progress!</span>")
+				return
+		active = TRUE
+
+		var/area/A = get_area(bombturf)
+		to_chat(user, "<span class = 'warning'>Calibrating GPS, remain still to continue with strike.</span>")
```

Comment:
Need to fix the grammar: "with the strike"

Reply:
Personally `with strike` seems correct, while the the would also work.